lib/vuln/php_easter_egg/engine.py

```diff
@@ -0,0 +1,201 @@
+                   thread_tmp_filename, extra_requirements, socks_proxy, scan_id, scan_cmd):
+    try:
+        try:
+            s = conn(target, port, timeout_sec, socks_proxy)
+        except Exception:
+            return False
+        if not s:
+            return False
+        else:
+            user_agent_list = useragents.useragents()
+            global php_easter_egg
+            if target_type(target) != "HTTP" and port == 443:
+                target = 'https://' + target
+            if target_type(target) != "HTTP" and port == 80:
+                target = 'http://' + target
+            if(not target.endswith("/")):
+                req_php_credits_url = target + "/" + extra_requirements["php_easter_eggs"][0]
+                req_php_logo_url = target + "/" + extra_requirements["php_easter_eggs"][1]
+                req_zend_url = target + "/" + extra_requirements["php_easter_eggs"][2]
+                req_php_logo2_url = target + "/" + extra_requirements["php_easter_eggs"][3]
+            try:
+                user_agent = {'User-agent': random.choice(user_agent_list)}
+                req_php_credits = requests.get(req_php_credits_url, timeout=10, headers=user_agent)
+                user_agent = {'User-agent': random.choice(user_agent_list)}
+                req_php_logo = requests.get(req_php_logo_url, timeout=10, headers=user_agent)
+                user_agent = {'User-agent': random.choice(user_agent_list)}
+                req_zend = requests.get(req_zend_url, timeout=10, headers=user_agent)
+                user_agent = {'User-agent': random.choice(user_agent_list)}
+                req_php_logo2 = requests.get(req_php_logo2_url, timeout=10, headers=user_agent)
+            except requests.exceptions.RequestException: 
+               return False
+            if req_php_credits.status_code == 200 and 'PHP Credits'.lower() in req_php_credits.text.lower():
+                php_easter_egg = "phpinfo() discosed"
+                return True
+            elif req_php_logo.status_code == 200 and "GIF89a".lower() in req_php_logo.text.lower():
+                php_easter_egg = "Php version disclosed which varies with the logo leaked"
+                return True
+            elif req_zend.status_code == 200 and "GIF89a".lower() in req_zend.text.lower():
+                php_easter_egg = "PHP Zend engine detected"
+                return True
+            elif req_php_logo2.status_code == 200 and "GIF89a".lower() in req_php_logo2.text.lower():
+                php_easter_egg = "PHP easter egg detected"
+                return True
+            else:
+                return False
+    except Exception as e:
+        print(e)
+        return False
+
+
+def __php_easter_eggs_vuln(target, port, timeout_sec, log_in_file, language, time_sleep,
+                     thread_tmp_filename, extra_requirements, socks_proxy, scan_id, scan_cmd):
+    if php_easter_eggs_vuln(target, port, timeout_sec, log_in_file, language, time_sleep,
```

Comment:
@aman566 please use the same name you started with 'php_easter_egg_vuln_scan' (not 'eggs')